core/templates/domain/question/editable-question-backend-api.service.ts

```diff
@@ -17,213 +17,273 @@
  *  backend.
  */
 
-require('domain/collection/guest-collection-progress.service.ts');
-require('domain/exploration/AnswerGroupObjectFactory.ts');
-require('domain/exploration/HintObjectFactory.ts');
-require('domain/exploration/OutcomeObjectFactory.ts');
-require('domain/exploration/ParamSpecObjectFactory.ts');
-require('domain/question/QuestionObjectFactory.ts');
-require('domain/exploration/WrittenTranslationObjectFactory.ts');
-require('domain/objects/FractionObjectFactory.ts');
-require('domain/utilities/url-interpolation.service.ts');
-
-require('domain/question/question-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('EditableQuestionBackendApiService', [
-  '$http', '$q', 'QuestionObjectFactory',
-  'UrlInterpolationService',
-  'EDITABLE_QUESTION_DATA_URL_TEMPLATE', 'QUESTION_CREATION_URL',
-  'QUESTION_SKILL_LINK_URL_TEMPLATE',
-  function(
-      $http, $q, QuestionObjectFactory,
-      UrlInterpolationService,
-      EDITABLE_QUESTION_DATA_URL_TEMPLATE, QUESTION_CREATION_URL,
-      QUESTION_SKILL_LINK_URL_TEMPLATE) {
-    var _createQuestion = function(
-        skillIds, skillDifficulties,
-        questionDict, imagesData, successCallback, errorCallback) {
-      var postData = {
-        question_dict: questionDict,
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { HttpClient } from '@angular/common/http';
+import { QuestionObjectFactory, QuestionBackendDict, Question } from 'domain/question/QuestionObjectFactory';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { QuestionDomainConstants } from 'domain/question/question-domain.constants';
+
+export interface EditableQuestionBackendResponse {
```

Comment:
@srijanreddy98 This can be made a Record<string, string> wherever applicable instead of making an interface?

Reply:
No, we use Record when we don't know the structure. Here we know what it is. So not required.

Reply:
okay! thanks for the clarification @srijanreddy98